src/main/kotlin/org/rust/cargo/runconfig/run/CargoRunProducer.kt

```diff
@@ -0,0 +1,47 @@
+package org.rust.cargo.runconfig.run;
+
+import com.intellij.execution.actions.ConfigurationContext
+import com.intellij.execution.actions.RunConfigurationProducer
+import com.intellij.openapi.util.Ref
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.impl.RustFnItemImpl
+
+class CargoRunProducer : RunConfigurationProducer<CargoRun>(CargoRunType()) {
+    override fun isConfigurationFromContext(configuration: CargoRun?, context: ConfigurationContext?): Boolean {
+        return configuration?.name == "Run " + context?.project?.name
+    }
+
+    override fun setupConfigurationFromContext(configuration: CargoRun?,
+                                               context: ConfigurationContext?,
+                                               sourceElement: Ref<PsiElement>?): Boolean {
+        if (isMainFile(context?.psiLocation?.containingFile)) {
+            configuration?.name = "Run " + context?.project?.name
+            return true
+        }
+        return false
+    }
+
+    private fun isMainFile(file: PsiFile?): Boolean {
+        if (file == null) {
+            return false
+        }
+
+        return file.name == "main.rs" && file.parent?.name == "src" && fileContainsMainFn(file)
```

Comment:
This check can be made more precise. There is `module.cargoProject`, which stores the information about the project as cargo sees it. In particular, file is a `MainFile` if it is a binary target. No need to fix it in this PR, the proper check can be implemented later.